src/opm/output/eclipse/InteHEAD.cpp

```diff
@@ -910,14 +970,47 @@ namespace {
         return maxNumActiveConn;
     }
 
+    template <typename AquiferCollection>
+    int maxAquID(const AquiferCollection& aquiferCollection)
+    {
+        return std::accumulate(aquiferCollection.begin(), aquiferCollection.end(), 0,
+                               [](const int maxID, const auto& aquiferData)
+                               {
+                                   return std::max(maxID, aquiferData.aquiferID);
+                               });
+    }
```

Comment:
Interesting and works. But wouldn't using
```
if (aquiferCollection.empty()) return 0;
return std::max_element(aquiferCollection.begin(), aquiferCollection.end(), 0,
                               [](const auto& aquiferData1, const auto& aquiferData2)
                               {
                                   return  aquiferData1.aquiferID < aquiferData2.aquiferID;
                               }) -> aquiferID;
```
work and be more explicit?

Reply:
> wouldn't using [`max_element(...)->aquiferID`] work and be more explicit?

It would definitely work, but I'm not a huge fan of the trailing iterator access (`->aquiferID`).  In fact I did not even see it when first reading the statement.  I guess I'm just more familiar with using reductions for this type of application.